src/cluster.h

```diff
@@ -152,7 +152,7 @@ typedef struct clusterState {
     clusterNode *importing_slots_from[CLUSTER_SLOTS];
     clusterNode *slots[CLUSTER_SLOTS];
     uint64_t slots_keys_count[CLUSTER_SLOTS];
-    rax *slots_to_keys;
+    dictEntry *slots_to_keys[CLUSTER_SLOTS];
```

Comment:
i'd expect somewhere to describe this (a bit hackish) solution in detail, but at the very least, let's add a comment here.
i.e. something like "the head of a linked list dictEntries (of keys) in each hash slot".

Reply:
i see it is described in `dictEntryMetadataSize`, maybe a reference here to `clusterDictEntryMetadata` will help?

Reply:
I think the "Slot to Key API" section in db.c would be a good place to document it in detail. How about moving this section (including the function slotToEntryAdd, etc.) to cluster.c?

Reply:
i'm ok with moving them.. we need to decide if that part is more about the database or about the cluster.
but note that `dbBackup` also has a similar reference.
maybe we can extract some interface and remove that coupling.
maybe we wanna try to remove the `include "cluster.h" in db.c and see where it takes us...
i'm not sure..

Reply:
If I move these functions to cluster.c, we definitely need cluster.h in db.c...

Reply:
I've moved the slot-to-key API to cluster.c and the dict metadata struct to cluster.h. In db.c, only calls the slot-to-key API are used.